solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -3187,20 +3186,29 @@ public void cleanupOldIndexDirectories(boolean reload) {
     }
   }
 
-  @SuppressWarnings({"rawtypes"})
-  private static final Map implicitPluginsInfo = (Map) Utils.fromJSONResource("ImplicitPlugins.json");
+  private static final class ImplicitHolder {
```

Comment:
It appears this is here to lazy-load?  But why lazy-load (added complexity of ImplicitHolder wrapper) -- this cost is eventually inescapable since these are needed to load any core, not some obscure/optional feature.
@madrob

Reply:
It was always eager reading from disk and lazy copying the plugins and a messy combination of the two. Now it’s lazy everything the first time, but yea probably could be eager. I don’t think we’re putting in too much complexity this way tho

Reply:
I have no problem with this being eager loaded